File: kakomon10/Java/q10.java
import java.util.*;

class Main {
  public static void main(String[] args) {
    Scanner sc = new Scanner(System.in);
    int N = Integer.parseInt(sc.next());
    List<Integer> t = new ArrayList<Integer>(N + 1);
    List<Integer> x = new ArrayList<Integer>(N + 1);
    List<Integer> y = new ArrayList<Integer>(N + 1);
    t.add(0);
    x.add(0);
    y.add(0);

    boolean can = true;
    for (int i = 0; i < N; i++) {
      t.add(Integer.parseInt(sc.next()));
      x.add(Integer.parseInt(sc.next()));
      y.add(Integer.parseInt(sc.next()));
    }
    sc.close();
    for (int i = 0; i < N; i++) {
      int dt = t.get(i + 1) - t.get(i);
      int dist = Math.abs(x.get(i + 1) - x.get(i)) + Math.abs(y.get(i + 1) - y.get(i));
      if (dt < dist) {
        can = false;
      }
      if (dist % 2 != dt % 2) {
        can = false;
      }
    }
    if (can) {
      System.out.println("Yes");
    } else {
      System.out.println("No");
    }
  }
}

File: kakomon10/py/q10.py
N = int(input())
x = list(range(N+1))
y = list(range(N+1))
t = list(range(N+1))
for i in range(1,N+1):
  t[i], x[i], y[i] = map(int, input().split())

# dt = t[i+1]-t[i]
# dist = abs(x[i+1]-x[i]) + abs(y[i+1]+y[i])
# このとき dist<=dt （t+1で動ける範囲はx方向y方向のどちらかに1のため）
# dtが偶数ならばdistも偶数である
# 例.dt=2のとき、distの取りうる値は 0, 2
can = True
for i in range(N):
  dt = t[i+1] - t[i]
  dist = abs(x[i+1]-x[i]) + abs(y[i+1]-y[i])
  if dt < dist :
    can = False
  if dt % 2 != dist % 2:
    can = False

if can :
  print("Yes")
else:
  print("No")